federated_query/processor/query_preprocessor.py

```
@@ -0,0 +1,392 @@
+            return sources
+        sources.append(self._build_source(from_clause.this))
+        joins = select.args.get("joins")
+        if joins:
+            for join in joins:
+                sources.append(self._build_source(join.this))
+        return sources
+
+    def _build_source(self, table_expr: exp.Expression) -> _SelectSource:
+        """Build metadata for a FROM/JOIN table."""
+        if not isinstance(table_expr, exp.Table):
+            raise StarExpansionError("Star expansion only supports base tables")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)